app/controllers/api/v3/courses_controller.rb

```diff
@@ -3,14 +3,25 @@ module V3
     class CoursesController < API::V3::ApplicationController
       before_action :build_recruitment_cycle
       before_action :build_provider
+      before_action :build_courses
 
       def index
-        @courses = @provider.courses
-        render jsonapi: @courses, include: params[:include]
+        links = {}
+        page_index = params.dig(:page, :page)
+        per_page = params.dig(:page, :per_page)
+
+        if per_page.present?
```

Comment:
I don't think paging should be optional ... we should default `per_page` to a number from settings (so it can be over-ridden just in case). 100 might be a reasonable start (seems to be the built-in default).